### PR TITLE
Allow for chaining _AnimationBuilder

### PR DESCRIPTION
**Pull request by eulertour**
## Motivation

Allow for chaining methods passed to `Scene.play()`

## Overview / Explanation for Changes

#885 

## Oneline Summary of Changes

```
- Allow for chaining methods passed to Scene.play()
```

## Testing Status

Added AnimationBuilderTest

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [x] Newly added functions/classes are either private or have a docstring
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [x] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

File: manim/mobject/mobject.py
Comment (friedkeenan) [inline manim/mobject/mobject.py:1318]: This shouldn't be necessary anymore, correct? Or perhaps it would be better to have a separate method within `_AnimationBuilder` for creating the animation that could be called in `Scene.compile_animations`? That would better organize the code I think, though this local import would still be a thing

File: manim/mobject/mobject.py
Comment (friedkeenan) [inline manim/mobject/mobject.py:1320]: Does this need to be an attribute of `self`? If not then I'd rather it just be `method = ...`

**Review by friedkeenan — COMMENTED** [review]
Disappointed that it can't just return an animation anymore, but oh well I suppose. These changes look good just have a couple small things I wanted to address

File: manim/mobject/mobject.py
Comment (behackl) [reply manim/mobject/mobject.py:1318]: This would probably also create a cleaner interface for custom animations, +1 for a dedicated method creating the animation.

**Comment by behackl** [comment]
@eulertour I wanted to push a commit improving documentation, but you have edits by org members disabled for this PR. Could you please either enable it, or apply my patch manually?

[0001-expand-animate-docstring-fix-some-examples-with-simu.zip](https://github.com/ManimCommunity/manim/files/5756898/0001-expand-animate-docstring-fix-some-examples-with-simu.zip)

File: manim/scene/scene.py
Comment (friedkeenan) [inline manim/scene/scene.py:621]: This should be checking `isinstance`, no?
```suggestion
            if isinstance(arg, _AnimationBuilder):
```

Comment (eulertour) [reply manim/scene/scene.py:621]: why?

**Comment by eulertour** [comment]
@behackl The box is actually checked so it might require more permissions or something

File: manim/scene/scene.py
Comment (friedkeenan) [reply manim/scene/scene.py]: Doesn't seem very pythonic to be checking type equality, particularly because that locks out any potential subclass of `_AnimationBuilder` from working when it could really work just fine. Part of pythonicity imo is giving developers lots of freedom/control (usually conveyed with the phrase "We're all consenting adults here"), and allowing subclasses gives more freedom and control.

**Comment by behackl** [comment]
> @behackl The box is actually checked so it might require more permissions or something

Weird, I actually can commit via the web interface, but it didn't let me push when I tried it on my machine yesterday. Anyhow, not so important -- thanks for applying the patch; I've fixed another call to animate in the latest commit and am waiting for the tests to pass before I approve; thank you for implementing this!

**Review by behackl — APPROVED** [review]

**Comment by behackl** [comment]
@friedkeenan, do you have any further comments on this?

**Review by friedkeenan — APPROVED** [review]
Woops yeah guess I forgot to approve, looks good to me

**Pull request merged**
